packages/angular/cli/commands/add-impl.ts

```diff
@@ -235,6 +231,28 @@ export class AddCommand extends SchematicCommand<AddCommandSchema> {
     return this.executeSchematic(collectionName, options['--']);
   }
 
+  private async isProjectVersionValid(packageIdentifier: npa.Result): Promise<boolean> {
```

Comment:
This was refactored into a separate method to avoid the 200 line limit for the `run` method.